trie/snap_difflayer.go

```diff
@@ -0,0 +1,231 @@
+// Copyright 2022 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package trie
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// diffLayer represents a collection of modifications made to the in-memory tries
+// after running a block on top.
+//
+// The goal of a diff layer is to act as a journal, tracking recent modifications
+// made to the state, that have not yet graduated into a semi-immutable state.
+type diffLayer struct {
```

Comment:
WHy is this file called `snap_difflayer.go` ? Instead of `trie_difflayer.go` or `difflayer.go` ?

Reply:
I want to group all PBSS relevant files with the prefix `snap`. :)

Reply:
Ah.. Yeah it's a bit unfurtunate in this specific case, since the other `difflayer` is in `core/state/snapshot` and handles `snap` data, as opposed to this one which deals with `trie` data :)

Reply:
A better solution is to move all trie database files to a separate package. 

It's possible if we can untie trie and trie database first. Specifically, https://github.com/ethereum/go-ethereum/pull/26813 is required.